packages/flame/lib/src/game/game_widget/game_widget.dart

```diff
@@ -135,10 +138,15 @@ class _GameWidgetState<T extends Game> extends State<GameWidget<T>> {
 
   MouseCursor? _mouseCursor;
 
-  Future<void>? _gameLoaderFuture;
-
-  Future<void> get _gameLoaderFutureCache =>
-      _gameLoaderFuture ?? (_gameLoaderFuture = widget.game.onLoad());
+  Future<void> get _loaderFuture {
+    final onLoad = widget.game.onLoadCache;
+    final onMount = widget.game.onMount;
+    if (onLoad != null) {
+      return onLoad.then((_) => onMount());
+    } else {
+      return Future<void>.value().then((_) => onMount());
```

Comment:
There must be a better way than this to encapsulate a void function call in a Future?

Reply:
I guess you could avoid `if` using the `??` operator:
```
(onLoad ?? Future<void>.value()).then((_) => onMount());
```

Reply:
Then `onMount` wouldn't be called when `onLoad` is defined. Or maybe I misunderstood your suggestion?

I want to chain them when onLoad is defined and only call `onMount` if `onLoad` is not defined.

What my comment was about was removing the empty future with the `then` and encapsulating `onMount` differently in a future, if it is possible.

Reply:
Ooh I missed the parenthesis on the phone, that is a good suggestion! I still want to remove the weird empty Future though.

Reply:
How about `Future(onMount)`?

Reply:
Lol, I thought I tried that! Thanks